extension/src/XliffEditor/XliffEditorPanel.ts

```diff
@@ -98,28 +101,31 @@ export class XliffEditorPanel {
                         }
                         return;
                     case "update":
+                        let suggestedTranslations: { id?: string, targetText: string }[] = [];
                         vscode.window.showInformationMessage(message.text);
+                        let targetUnit = this._xlfDocument.getTransUnitById(message.transunitId);
                         this._xlfDocument.getTransUnitById(message.transunitId).targets[0].textContent = message.targetText;
                         this._xlfDocument.getTransUnitById(message.transunitId).targets[0].translationToken = undefined;
+                        this._xlfDocument.getTransUnitById(message.transunitId).insertCustomNote(CustomNoteType.RefreshXlfHint, "Translated with Xliff Editor");
+                        this._xlfDocument.transunit.filter(a => (a.source === targetUnit.source) && !a.identicalTargetExists(message.targetText) && (a.id !== targetUnit.id) /*&& a.hasCustomNote(CustomNoteType.RefreshXlfHint)*/).forEach(unit => {
+                            let suggestion = new Target(message.targetText);
+                            suggestion.translationToken = TranslationToken.Suggestion;
+                            if (unit.targets.length === 1 && (unit.targets[0].textContent === "")) {
+                                unit.targets[0] = suggestion;
+                                suggestedTranslations.push({ id: unit.id, targetText: suggestion.textContent });
+                            }
+                        });
```

Comment:
I think this block could be extracted to its own function.

I believe we'll have to add some more logic here as well... 
1. Check if there are any trans-units with same source, with suggestion token and target same as current target was before editing -> Change suggestion to message.targetText. Would happen if you first translate the target, moves out of textbox, and then focus on the textbox again and change anything.
2. If above occurs and the new target is empty, I think we'll remove the suggestion for the other transunits.

This could also solve the undo scenario....

Reply:
I'm not quite following this, I think we need to talk it through :)

Reply:
The part I thought was in need of it's own function was the suggestion part... ;)
The scenario I want to cover is when you write a translation (that adds suggestions) and then you move out of the textbox, unchecking "Completed", translate the next trans-unit (or whatever), then move back to the first translation and change something. Then you want the old suggestion to change as well.

But we can talk tomorrow :)